pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -383,6 +383,16 @@ func (r *ReconcileClusterPool) Reconcile(ctx context.Context, request reconcile.
 			log.WithError(err).Error("error adding clusters")
 			return reconcile.Result{}, err
 		}
+	// Special case for stale CDs: allow deleting one if all CDs are installed.
+	case drift == 0 && len(cds.Installing()) == 0 && len(cds.Stale()) > 0:
```

Comment:
Actually does this need a check to make sure none are deleting? Otherwise it kinda looks like it might immediately wipe them all out.

Reply:
It shouldn't need that:
- On this reconcile we `Delete` one.
- The next reconcile will miss this `case` because we'll be one short of `pool.size` so `drift` will be `< 0`. It'll add a CD  via L382 which will go to `Installing`.
- The next reconcile (and subsequent ones for the next 40m) will see nonzero `Installing` clusters, so will skip this `case`.

Did I logic that right?

Reply:
You are probably right, there are cases where the cache isn't updated for new objects and may not see the new Installing cluster, which is where we have to use expectations. If I'm correct that *might* happen here, it's possible it could wipe out a couple at once before the installing start showing up. I'm not super confident in any of this so we can see how it goes.

Reply:
Could we do a counter metric here so we can see the rate at which clusters are getting replaced due to pool changes.

Reply:
Counters are since the process started, prometheus handles the past data and adds the two together. In this case it's not so much for the value of the counter, more so for the rate at which it's occurring.

A gauge does sound useful for the total number of stale at any given time, rather than the rate at which we're doing replacements.